Mathlib/CategoryTheory/SingleObj.lean

```diff
@@ -212,6 +212,26 @@ theorem id_toFunctor : (id M).toFunctor = 𝟭 _ :=
 
 end MonoidHom
 
+namespace SingleObj
+
+variable {M : Type u} {N : Type v} [Monoid M] [Monoid N]
+
+/-- Reinterpret a monoid isomorphism `f : M ≃* N` as an equivalence `SingleObj M ≌ SingleObj N`. -/
+@[simps!]
+def toSingleObjEquiv (e : M ≃* N) : SingleObj M ≌ SingleObj N := by
+  apply CategoryTheory.Equivalence.mk
+  show _ ≅ MonoidHom.toFunctor e.toMonoidHom ⋙ MonoidHom.toFunctor e.symm.toMonoidHom
+  dsimp [MonoidHom.toFunctor, SingleObj.mapHom, Functor.comp]
+  apply eqToIso
+  simp only [MulEquiv.symm_apply_apply]
+  rfl
+  dsimp [MonoidHom.toFunctor, SingleObj.mapHom, Functor.comp]
+  apply eqToIso
+  simp only [MulEquiv.apply_symm_apply]
+  rfl
```

Comment:
It would be more readable as follows:
```suggestion
def toSingleObjEquiv (e : M ≃* N) : SingleObj M ≌ SingleObj N where
  functor := e.toMonoidHom.toFunctor
  inverse := e.symm.toMonoidHom.toFunctor
  unitIso := eqToIso (by
    rw [← MonoidHom.comp_toFunctor, ← MonoidHom.id_toFunctor]
    congr 1
    aesop_cat)
  counitIso := eqToIso (by
    rw [← MonoidHom.comp_toFunctor, ← MonoidHom.id_toFunctor]
    congr 1
    aesop_cat)
```